lib/ddtrace/contrib/http/patcher.rb

```diff
@@ -110,12 +110,18 @@ def request(req, body = nil, &block) # :yield: +response+
 
               pin.tracer.trace(NAME) do |span|
                 begin
+                  if req.respond_to?(:uri) && req.uri
+                    host_address = req.uri.host
+                    host_port = req.uri.port.to_s
+                  else
+                    host_address = @address
+                    host_port = @port.to_s
+                  end
+
                   span.service = pin.service
                   span.span_type = Datadog::Ext::HTTP::TYPE
 
-                  span.resource = req.method
-                  # Using the method as a resource, as URL/path can trigger
-                  # a possibly infinite number of resources.
+                  span.resource = host_address
```

Comment:
This will be great for the majority of use-cases. It can be common to have multiple endpoints on localhost (e.g. using haproxy to load balance traffic, or multiple services such as elastic search).

Possibly the resource could be set to `span.resource = "#{host_address}:#{host_port}"` ?